controllers/internal/multierror.go

```diff
@@ -0,0 +1,25 @@
+package internal
+
+import (
+	"strings"
+)
+
+// NewMultiErrorWithNewLineSeparator creates a MultiError that uses "\n" as separator for each error.
+func NewMultiErrorWithNewLineSeparator(errs []error) error {
+	return &MultiError{Errors: errs, Separator: "\n"}
```

Comment:
How does this look in the logs? I wonder if using a comma instead would be a better choice. Wdyt?

Reply:
No, I'm ok with this if you are happy with how it looks in the logs.

Reply:
Changed it to ; since I noticed we do use JSON logging.